### PR TITLE
standard storage path migration.

### PR DESCRIPTION
**Pull request by jortel**
https://pulp.plan.io/issues/1819

After the mongoengine conversion, python units stopped being created with storage_path that ended with the package filename.  Discussed with @rbarlow and we decided to fix this as part of this migration.

### CODE REVIEW

File: plugins/pulp_python/plugins/importers/sync.py [hunk 1]
```diff
@@ -144,8 +144,11 @@ def download_succeeded(self, report):
             return self.download_failed(report)
 
         package = models.Package.from_archive(report.destination)
+        package.set_storage_path(os.path.basename(report.destination))
```
Comment (mhrivnak) [inline plugins/pulp_python/plugins/importers/sync.py]: Is this change related to the migration?

Comment (mhrivnak) [inline plugins/pulp_python/plugins/importers/sync.py]: I see, it looks like you're correcting a storage path bug in the migration and in code, which does make sense to go together.

File: plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        :type unit: pulp.plugins.migration.standard_storage_path.Unit
+        :return: The new path.
+        :rtype: str
+        """
+        filename = unit.document['_filename']
+        path = unit.document['_storage_path']
+        if not path.endswith(filename):
+            unit.document['_storage_path'] = filename
+        new_path = super(Package, self)._new_path(unit)
+        return new_path
```
Comment (jortel) [inline plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py]: @rbarlow any idea why I'm gettting:
```
[jortel@f23d pulp_python]$ ./run-tests.py 
checking pep257 for failures, ignoring D100,D103,D104,D200,D202,D203,D205,D400,D401,D402
Function _new_path is missing sphinx type definition for argument unit (./plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py:27)
found undocumented parameters!
```
I may be staring right at it and not seeing the problem.  Need a 2nd pair of eyes.

File: plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        self.fields.add('_filename')
+
+    def _new_path(self, unit):
+        """
+        Units created by 2.8.0 don't include the filename.  This was a regression
+        that is being corrected by this additional logic.  If the storage path
+        does not end with the filename stored in the unit, it is appended.
+
+        :param unit: The unit being migrated.
+        :type unit: pulp.plugins.migration.standard_storage_path.Unit
```
Comment (jortel) [inline plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py]: @rbarlow any idea why I'm gettting:
```
[jortel@f23d pulp_python]$ ./run-tests.py 
checking pep257 for failures, ignoring D100,D103,D104,D200,D202,D203,D205,D400,D401,D402
Function _new_path is missing sphinx type definition for argument unit (./plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py:27)
found undocumented parameters!
```
I may be staring right at it and not seeing the problem.  Need a 2nd pair of eyes.

Comment (rbarlow) [inline plugins/pulp_python/plugins/migrations/0001_standard_storage_path.py]: @jortel Hmmmm, that is very odd! Unfortunately, I don't have any ideas. Does anyone else have an idea what the problem might be here?

**Comment by seandst** [comment]
ok test

**Comment by jortel** [comment]
ok test
